pkg/daemon/ceph/osd/volume.go

```diff
@@ -853,6 +853,59 @@ func (a *OsdAgent) appendDeviceClassArg(device *DeviceOsdIDEntry, args []string)
 	return args
 }
 
+// WipeDevicesFromOtherClusters wipes the OSD backed disks if they have metadata from a different ceph cluster.
+// The wiped disks can then be used to prepare OSDs for the current ceph cluster.
+func (a *OsdAgent) WipeDevicesFromOtherClusters(context *clusterd.Context) error {
+	args := []string{"raw", "list", "--format", "json"}
+
+	result, err := callCephVolume(context, args...)
+	if err != nil {
+		return errors.Wrapf(err, "failed to retrieve ceph-volume raw list results")
+	}
+
+	var cephVolumeResult map[string]osdInfoBlock
+	err = json.Unmarshal([]byte(result), &cephVolumeResult)
+	if err != nil {
+		return errors.Wrapf(err, "failed to unmarshal ceph-volume raw list results")
+	}
+
+	if len(cephVolumeResult) == 0 {
+		logger.Info("No existing OSDs were found. No disks to be wiped")
+		return nil
+	}
+
+	for _, result := range cephVolumeResult {
+		// Wipe the devices that will be used for preparing OSD but already have metadata from another ceph cluster
+		if result.CephFsid != a.clusterInfo.FSID {
+			if isDesiredDevice(result.Device, context.Devices) {
+				logger.Infof("BEGIN wiping OSD %d device belonging to a different ceph cluster %q ", result.OsdID, result.CephFsid)
+				logger.Infof("zap OSD.%d path %q", result.OsdID, result.Device)
+				output, err := context.Executor.ExecuteCommandWithOutput("stdbuf", "-oL", "ceph-volume", "lvm", "zap", result.Device, "--destroy")
+				if err != nil {
+					return errors.Wrapf(err, "failed to zap osd.%d path %q. %s.", result.OsdID, result.Device, output)
+				}
+				logger.Infof("%s\n", output)
```

Comment:
```suggestion
				logger.Infof("%s", output)
```
do we need extra new line, give `logger.Infof` comes with new line?

Reply:
done.